### PR TITLE
ROX-19613: Retry http calls in roxctl

### PR DESCRIPTION
**Pull request by dhaus67**
## Description

We've been dealing with issues within roxctl when calling APIs via HTTP. For gRPC, retries have already been added, however for HTTP not yet.

This PR adds retries to the HTTP client shared for each command within the environment by leveraging [go-retryablehttp](https://github.com/hashicorp/go-retryablehttp).

The current retry logic is a maximum of 3 retries, using an exponential backoff with a minimum wait time of 10 second and a maximum wait time of 30 seconds in-between requests. This should help with CI flakes we've seen.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

- see CI.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Comment by dhaus67** [comment]
Current dependencies on/for this PR:
* master
  * **PR #7801** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/7801" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/7801?utm_source=stack-comment).

**Comment by stehessel** [comment]
What is the actual total retry time? 1s, 2s, 4s? I would propose to extend the total retry time to about ~1 minute to realistically cover a Central restart window.

**Comment by dhaus67** [comment]
@stehessel adjusted the starting waiting time to 10 seconds, leads to a total waiting time of 10 + 20 + 30 (due to max wait time being capped at 30s) = 60s with the current retry configuration.

File: central/scannerdefinitions/handler/handler.go [hunk 1]
```diff
@@ -261,7 +261,7 @@ func (h *httpHandler) post(w http.ResponseWriter, r *http.Request) {
 	}
 
 	if err := h.handleZipContentsFromVulnDump(r.Context(), tempFile); err != nil {
-		httputil.WriteGRPCStyleError(w, codes.Internal, err)
```
Comment (stehessel) [inline central/scannerdefinitions/handler/handler.go:264]: Why this change?

Comment (dhaus67) [reply central/scannerdefinitions/handler/handler.go:264]: Previousy, `codes.Internal` lead to a 500 HTTP status code, which the retryable client will try to retry on (since per definition, this doesn't really mean that anything was _wrong_ with the provided data but rather a server issue).

While we could adjust the retry policy, this felt like a wrong status code on the backend side: we return an error because the provided file wasn't of ZIP format, so invalid argument was a more appropriate status code to return.

Comment (stehessel) [reply central/scannerdefinitions/handler/handler.go:264]: Are you sure the retry handler would retry a POST by default? It's not guaranteed to be idempotent. But I get your point, it wasn't obvious to me that the error here is user induced.

**Review by stehessel — APPROVED** [review]

File: central/scannerdefinitions/handler/handler.go (hunk 1 above)
Comment (dhaus67) [reply central/scannerdefinitions/handler/handler.go:264]: Default behavior of the client is to retry irrespective of the HTTP method.

We may adjust this going forward, but for the time being should be OK.
